### PR TITLE
Updated OWNERS file to reflect prow OWNERS file

### PR DESCRIPTION
**Pull request by dislbenn**
Signed-off-by: dislbenn <dbennett@redhat.com>

**Related Issue:**  stolostron/backlog#<ISSUE_NUMBER>

### CODE REVIEW

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

No need to duplicate approvers names in reviewers.

**Comment by dislbenn** [comment]
Related PR: https://github.com/openshift/release/pull/31819

**Pull request merged**
